parquet-hadoop/src/main/java/org/apache/parquet/crypto/AesCipher.java

```diff
@@ -68,19 +67,32 @@
 
   public static byte[] createModuleAAD(byte[] fileAAD, ModuleType moduleType, 
       short rowGroupOrdinal, short columnOrdinal, short pageOrdinal) {
```

Comment:
The links to the discussion on this,
https://github.com/apache/parquet-format/pull/114#discussion_r234022579
https://github.com/apache/parquet-format/pull/114#discussion_r232941138
http://mail-archives.apache.org/mod_mbox/parquet-dev/201901.mbox/%3CCAO4re1kM4xGMNT4CGrjvA43t-QgUmUwLMskTJfd8ivgCfF8rSw%40mail.gmail.com%3E

The parquet-cpp approach to this is to allow for any number of row groups in files without encryption, and to limit it to 32K in encrypted files,
https://github.com/apache/arrow/commit/0c5168cf203ddf94dff01c178d649853323acbb8
"While writing files with so many row groups is a bad idea, people will still do it... This .. enables reading the many-row-group files again. Files with encrypted row group metadata with that many row groups cannot be read"

Reply:
Also - like with the page numbers in the previous comment, having too many row groups will adversely affect encryption performance. There are per-rowgroup encryption operations, always performed on small  buffers - therefore, very slow. Having dozens/hundreds of thousands (or more) of them will significantly affect the overall encryption time of a file. Moreover, having lots of row groups might lead to having smaller data pages, which decreases the performance further.

Reply:
Still, the variables in this function (and elsewhere) don't have to be `short`. After looking at the code, it seems `int`s are better suited for managing and checking these parameters (and for enabling any values in unencrypted files). I'll make this change.

Reply:
I understand that large numbers of pages/row groups or columns would lead to significant performance drawbacks but it should not limit what the spec allows otherwise.

Since it is discussed and approved already, I am fine with using `short` values for these. What I would suggest adding though is to have the conversion from `int` to `short` centralized and and have specific error messages so it is clear that the limit reached is a hard limit for the encryption feature. Also, if we will publish any description/example for the encryption feature these limitations shall be listed there.

One more thing: the check of `intValue > Short.MAX_VALUE` is not complete. In case of `intValue` is negative the cast may result in a valid positive `short` value. I would suggest using `(short) intValue != intValue` instead.

Reply:
Thanks for the value checking tip, I'll update the code to use it. As for a centralization - I think this function (`createModuleAAD`) is the right place. In the encryption feature, ordinals are used only for integrity verification - performed via AADs, which are calculated here for both encryption and decryption. Everywhere in the code, the ordinals will be an `int`. Since the `createModuleAAD` is called only for encrypted files, an exception will be thrown only for them (if an ordinal exceeds the `Short.MAX_VALUE`).

Reply:
Sounds good to me.